sdk/rust/oak_config_serializer/src/main.rs

```diff
@@ -78,13 +87,45 @@ impl Default for InitialNodeConfig {
     }
 }
 
+fn get_sha256(data: &[u8]) -> String {
+    let mut hasher = Sha256::new();
+    hasher.update(data);
+    hex::encode(hasher.finalize().as_slice().to_vec())
+}
+
 /// Load Wasm module from URL or a file, if URL is not specified.
-fn load_module(module: &Module) -> anyhow::Result<Vec<u8>> {
+async fn load_module(module: &Module) -> anyhow::Result<Vec<u8>> {
     match module {
-        Module::Url(_url) => unimplemented!(),
         Module::Path(path) => {
             fs::read(&path).with_context(|| format!("Couldn't read file {}", path))
         }
+        Module::External(external) => {
+            let url: Url = external.url
+                .parse()
+                .context("Couldn't parse URL")?;
+
+            debug!("Downloading module from: {}", url);
+            // TODO(#1240): Add a Wasm module cache.
+            let response = reqwest::get(url.clone())
```

Comment:
Why can `url` not be passed by reference?

Reply:
It's used in `.with_context(|| format!("Couldn't download module from {}", url))?;` that moves it in the context of lambda.